src/lavinmq/queue/queue.cr

```diff
@@ -840,7 +852,7 @@ module LavinMQ
 
     def rm_consumer(consumer : Client::Channel::Consumer, basic_cancel = false)
       deleted = @consumers.delete_consumer consumer
-      consumer_unacked_size = @unacked.sum { |u| u.consumer == consumer ? 1 : 0 }
+      consumer_unacked_size = @unacked.bytesize # { |u| u.consumer == consumer ? 1 : 0 }
```

Comment:
Keep the conditional sum here?

Reply:
Hmm, maybe I didn't understand the code good enough here ... :)

Reply:
As for my understanding the Unacked queue class have an internal unacked queue. Each queue item is a unack record containing segment position and a consumer.

```
unacked = @unacked
unack = Unack.new(sp, consumer)
if idx = unacked.bsearch_index { |u| u.sp > sp }
  unacked.insert(idx, unack)
else
  unacked << unack
end
```

The sum can the be feed a block, filtering out the consumer.
```
def sum(&blk : Unack -> _) : UInt64
  @unacked.sum(0_u64, &blk)
end
```

On the otherhand `consumer_unacked_size` are only used to log the message size of the unacked messages when removing a specific consumer. So perhaps not even needed.

Reply:
Thanks. Yeah, I think we'd like to keep the correct `consumer_unacked_size` logging there, but put the calculation into the log block so it's only triggered in debug log mode.